gradio/blocks.py

```diff
@@ -2532,6 +2533,12 @@ def reverse(text):
                 # In contrast, in the Wasm env, we can't do that because `threading` is not supported and all async tasks will run in the same event loop, `pyodide.webloop.WebLoop` in the main thread.
                 # So we need to manually cancel them. See `self.close()`..
                 self.run_startup_events()
+                # In the normal mode, self.run_extra_startup_events() is awaited like https://github.com/gradio-app/gradio/blob/2afcad80abd489111e47cf586a2a8221cc3dc9b6/gradio/routes.py#L1442.
+                # But in the Wasm env, we need to call the start up events here as described above, so we can't await it as here is not in an async function.
+                # So we use run_until_complete() instead. This is a best-effort fallback in the Wasm env but it doesn't guarantee that all the tasks are completed before they are needed.
+                asyncio.get_event_loop().run_until_complete(
```

Comment:
Can you use `get_running_loop` here?

Reply:
Thanks, good point.
Or higher level API `asyncio.create_task()` is more preferable?

Reply:
Replaced it to `asyncio.create_task()`. It is originally non-blocking so I think it's less confusing about its behavior.